src/data/repositories/ConfigD2Repository.ts

```diff
@@ -5,6 +5,7 @@ import { Region } from "$/domain/entities/Region";
 import { Future, FutureData } from "$/domain/entities/generic/Future";
 import { ConfigRepository } from "$/domain/repositories/ConfigRepository";
 import { D2Api } from "$/types/d2-api";
+import { extractFirstTwoLetters, extractPrefix } from "$/utils/string";
```

Comment:
Adding methods to some utils file in the root folder is only advisable for extremely generic functions. I don't think that's the case. The abstraction we are looking for is not "extractFirstTwoLetters", but some function that references the processing over the code (similar to the naming you had before).

This function is used in data and domain, right? So if this is something the domain must be aware of, let's move the function there, with another name (possible in Region and/or OrgUnit).